flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetMapInputFormat.java

```diff
@@ -40,10 +40,20 @@
  */
 public class ParquetMapInputFormat extends ParquetInputFormat<Map> {
 
+    /**
+     * @deprecated using this constructor is deprecated as the {@link MessageType} can be determined
+     *     from the file and does not need to be provided by the user. Prefer using the constructor
+     *     without MessageType parameter.
+     */
+    @Deprecated
     public ParquetMapInputFormat(Path path, MessageType messageType) {
         super(path, messageType);
```

Comment:
Since we do not check `messageType != null` anymore in base class, we need to add a `Preconditions#checkNotNull` here.